### PR TITLE
Add code changes to add more functionalities to ProjectsLibrary.

### PR DESCRIPTION
**Pull request by meghana87**
Merge pull request #5 from neeldhwaj/dev

Add ProjectsLibrary

### CODE REVIEW

File: contracts/MembersLibrary.sol [hunk 1]
```diff
@@ -65,13 +65,14 @@ library MembersLibrary {
 
     function getMember(address memberStoreAddress, uint index) constant returns (address account, uint state, uint dateAdded) {
         var memberStore = DataStore(memberStoreAddress);
-        if (index < 1) {
+        if (index < 1 || index > memberStore.count()) {
```
Comment (neeldhwaj) [inline contracts/MembersLibrary.sol]: This condition cannot be here as 'index' is the accountIndex(superset member index) which can be more than the memberIndex(subset member index).

File: contracts/MembersLibrary.sol [hunk 2]
```diff
@@ -50,7 +50,7 @@ library MembersLibrary {
         // mapping subset memberIndex to superset index
         memberStore.setIntIndex(sha3('memberIndex'), memberIndex, index);
         memberStore.setAddressValue(sha3('account', index), account);
-        // memberStore.setAddressIndex('account', account, index);
+        memberStore.setAddressIndex(sha3('account'), account, index);
```
Comment (neeldhwaj) [inline contracts/MembersLibrary.sol]: As proposed by you, updating sha3('account') to sha3('accountIndex'). We can also change the variable name index to accountIndex for more clarity.

File: contracts/MembersLibrary.sol [hunk 3]
```diff
@@ -65,13 +65,14 @@ library MembersLibrary {
...
         var memberStore = DataStore(memberStoreAddress);
-        if (index < 1) {
+        if (index < 1 || index > memberStore.count()) {
             return;
         }
         account = memberStore.getAddressValue(sha3('account', index));
         if (account == 0x0) {
             return;
         }
+        account = memberStore.getAddressValue(sha3('account', index));
```
Comment (neeldhwaj) [inline contracts/MembersLibrary.sol]: Taking from above, sha3('account', index) to sha3('accountIndex', index).

File: contracts/ProjectsLibrary.sol [hunk 4]
```diff
@@ -21,30 +21,30 @@ library ProjectsLibrary {
...
         projectStore.setAddressValue(sha3('projectOwner', projectIndex),msg.sender);
-        projectStore.setIntValue(sha3('projectBudet', projectIndex), budget);
+        projectStore.setIntValue(sha3('projectBudget', projectIndex), budget);
         projectStore.setIntValue(sha3('projectDateCreated', projectIndex), now);
         projectStore.setIntValue(sha3('projectStatus', projectIndex), 0);
     }
 
-    function getProject(address projectStoreAddress, uint projectID) constant returns (string projectTitle,
-                        string projectDescription, address projectOwner, uint projectBudet, uint projectMemberCount,
+    function getProject(address projectStoreAddress, uint projectIndex) constant returns (string projectTitle,
```
Comment (neeldhwaj) [inline contracts/ProjectsLibrary.sol]: We should remove the string returns from here as we cannot return dynamic sized data from one contract to another. The same has to be updated in Organisation.sol.

**Review by neeldhwaj — CHANGES_REQUESTED** [review]

File: contracts/MembersLibrary.sol (hunk 1 above)
Comment (meghana87) [reply contracts/MembersLibrary.sol]: Fixed.

File: contracts/MembersLibrary.sol (hunk 2 above)
Comment (meghana87) [reply contracts/MembersLibrary.sol]: Made the necessary code changes here.

File: contracts/MembersLibrary.sol (hunk 3 above)
Comment (meghana87) [reply contracts/MembersLibrary.sol]: Done.

File: contracts/ProjectsLibrary.sol (hunk 4 above)
Comment (meghana87) [reply contracts/ProjectsLibrary.sol]: This is fixed too.

**Pull request merged**